builtins/amp-ad.js

```diff
@@ -98,11 +99,31 @@ export function upgradeImages_(images) {
  * @return {undefined}
  */
 export function installAd(win) {
+
+  /**
+   * @type {boolean} Heuristic boolean as for whether another ad is currently
+   *     loading.
+   */
+  let loadingAdsCount = 0;
+
   class AmpAd extends BaseElement {
 
     /** @override  */
     renderOutsideViewport() {
-      return false;
+      // Before the user has scrolled we only render ads in view. This prevents
```

Comment:
Yep, its true that this needs to be dealt with in the viewer for real.

But I think for low priority content it is prudent to stay conservative with loading.